### PR TITLE
add fixpoint type to handle conditions

### PR DESCRIPTION
**Pull request by christian-lanius**
This pull request fixes issue #42 
However it misses a test for this. The previously failing excerpt from my sdf file can be found in the issue. I am unsure how to write set up a proper test for this. Doing it the "simple" way results in a comparison between two equivalent files, which have different white space however.

### CODE REVIEW

**Comment by mithro** [comment]
Hi,

Thank you for sending the pull request! It's awesome to see you contributing to this project. However, before I can look at this request you need to add [DCO sign-off](https://developercertificate.org/) to commits in this repository.

We require [DCO sign-off](https://developercertificate.org/) for **all** commits to this repository (including all authors). Please see the [following URL](https://github.com/probot/dco#how-it-works) to see more information and BeeWare also have a [Beginners guide to DCOs](https://pybee.org/contributing/how/dco/).

You can use `git commit --amend --signoff` and then a `git push --force` to add the signoff. You can also use `git rebase --signoff`.

Thanks!

**Comment by christian-lanius** [comment]
I did the DCO sign-off as requested and I also added a test case,  which fails in the current master branch.

**Comment by mithro** [comment]
@kgugala - Could you please review (or assign someone to review)?

File: sdf_timing/sdflex.py [hunk 1]
```diff
@@ -122,12 +123,17 @@
 
 
 # define FLOAT as function so it takes precendence over STRING
+def t_FIXPOINT(t):
+    r"[0-9]*'[bohd][0-9]+"
```
Comment (kgugala) [inline sdf_timing/sdflex.py:127]: hmm according to the sdf spec only binary constants are supprted (see http://ce.sharif.edu/courses/85-86/2/ce483/resources/root/sdf_3.0.pdf page 83:
```
SCALAR_CONSTANT ::= 1‘b0 // logical zero
                ||= 1‘b1 // logical one
                ||= 1‘B0 // logical zero
                ||= 1‘B1 // logical one
                ||= ‘b0 // logical zero
                ||= ‘b1 // logical one
                ||= ‘B0 // logical zero
                ||= ‘B1 // logical one
                ||= 0 // logical zero
                ||= 1 // logical one
```

**Review by kgugala — CHANGES_REQUESTED** [review]
@christian-lanius thanks for the PR. Generally looks good. Please see my comments

File: sdf_timing/sdflex.py (hunk 1 above)
Comment (christian-lanius) [reply sdf_timing/sdflex.py]: Renamed FIXPOINT into SCALARCONSTANT to better reflect this. I have also tightened the regex to better reflect that only binary constants are allowed. 

The commit ab3ba07 below passes the tests. I haven't worked with ply before, so on a more fundamental level, I am unsure if this is the "best" way to solve this. Technically, FLOAT are not allowed at this point in the as far as I can tell, thus we should get rid of it in "equation", then the SCALARCONSTANT has to match on 0 and 1 as well. However, this results in ambiguous grammar, which makes some of the tests fail. So instead, I left it as is.

File: sdf_timing/sdflex.py [hunk 2]
```diff
@@ -122,12 +123,17 @@
 
 
 # define FLOAT as function so it takes precendence over STRING
```
[the remark this answers is not in the document]
Comment (christian-lanius) [reply sdf_timing/sdflex.py]: Fixed (hopefully) by ab3ba07

**Review by kgugala — APPROVED** [review]

**Comment by kgugala** [comment]
LGTM. Thanks @christian-lanius

**Pull request merged**